src/Symfony/Component/TypeInfo/TypeResolver/StringTypeResolver.php

```diff
@@ -166,7 +170,8 @@ private function getTypeFromNode(TypeNode $node, ?TypeContext $typeContext): Typ
                 'static' => $typeContext ? Type::object($typeContext->getCalledClass()) : throw new InvalidArgumentException(\sprintf('A "%s" must be provided to resolve "static".', TypeContext::class)),
                 'parent' => $typeContext ? Type::object($typeContext->getParentClass()) : throw new InvalidArgumentException(\sprintf('A "%s" must be provided to resolve "parent".', TypeContext::class)),
                 'void' => Type::void(),
-                'never', 'never-return', 'never-returns', 'no-return' => Type::never(),
+                'never' => Type::never(),
+                'never-return', 'never-returns', 'no-return' => Type::never(new BoundBuiltinType(BoundBuiltinTypeIdentifier::tryFrom($node->name))),
```

Comment:
```suggestion
                'never', 'never-return', 'never-returns', 'no-return' => Type::never(),
```
I don't think there is any value to be this precise, `never` should be enough.